### PR TITLE
Unique

### PR DESCRIPTION
**Pull request by lutfuahmet**
Unique function for comparable types

### CODE REVIEW

File: slices_test.go [hunk 1]
```diff
@@ -216,3 +216,24 @@ func TestReverse(t *testing.T) {
 		}
 	})
 }
+
+func TestUnique(t *testing.T) {
+	t.Run("make unique", func(t *testing.T) {
```
Comment (isacikgoz) [inline slices_test.go:221]: Can we add a test case for `nil` or empty slice? Just to be on the safe side for future expansions.

**Review by isacikgoz — COMMENTED** [review]
Awesome addition @lutfuahmet 💯 Just a small request and question below, otherwise looks good to me.

File: slices.go [hunk 2]
```diff
@@ -66,10 +66,23 @@ func Shuffle[T any](s []T) []T {
...
 }
+
+// Unique removes duplicate values in a slice
+func Unique[T comparable](src []T) []T {
+	var result []T
+	var elemMap = make(map[T]bool)
+	for i := range src {
+		if !elemMap[src[i]] {
+			result = append(result, src[i])
+			elemMap[src[i]] = true
```
Comment (isacikgoz) [inline slices.go:84]: Would using empty struct be more efficient?

```suggestion
	var elemMap = make(map[T]struct{})
	for i := range src {
		if _, ok := elemMap[src[i]]; !ok {
			result = append(result, src[i])
			elemMap[src[i]] = struct{}{}
```

Comment (lutfuahmet) [reply slices.go]: copy that. I made those changes and pushed.
